### PR TITLE
fix: preserve loggers across migrations

### LINKED ISSUE #2825: verbose logging during migrations for troubleshooting.
As a user, I need to know how to log the details of the migration.

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #2825 

This fixes that logging was permanently broken post migrations. This also increases the log level while running migrations for debugging purposes.

Screenshot 2024-04-09 at 5 42 08 PM

### CODE REVIEW

File: src/phoenix/db/migrations/env.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # Interpret the config file for Python logging.
 # This line sets up loggers basically.
 if config.config_file_name is not None:
-    fileConfig(config.config_file_name)
+    fileConfig(config.config_file_name, disable_existing_loggers=False)
```
Comment (mikeldking) [inline src/phoenix/db/migrations/env.py:18]: this was sticking - causing errors to get stuck

**Review by anticorrelator — APPROVED** [review]

File: src/phoenix/db/alembic.ini [hunk 2]
```diff
@@ -104,7 +104,7 @@ handlers =
 qualname = sqlalchemy.engine
 
 [logger_alembic]
-level = WARN
+level = INFO
```
Comment (axiomofjoy) [inline src/phoenix/db/alembic.ini:107]: Curious, is there a way of setting the logging level via environment variables?

File: src/phoenix/db/migrate.py [hunk 3]
```diff
@@ -25,4 +47,7 @@ def migrate(url: URL) -> None:
     alembic_cfg.set_main_option("script_location", scripts_location)
     alembic_cfg.set_main_option("sqlalchemy.url", str(url))
 
-    command.upgrade(alembic_cfg, "head")
+    with using_log_level(logging.DEBUG, ["sqlalchemy", "alembic"]):
```
Comment (axiomofjoy) [inline src/phoenix/db/migrate.py:50]: It would be helpful to have this as an environment variable.

File: src/phoenix/db/migrate.py [hunk 4]
```diff
@@ -1,10 +1,31 @@
...
 from pathlib import Path
+from typing import Any, Generator
 
 from alembic import command
 from alembic.config import Config
 from sqlalchemy import URL
 
+
+@contextmanager
+def using_log_level(new_level: int, logger_names: list[str]) -> "Generator[Any, Any, Any]":
```
Comment (axiomofjoy) [inline src/phoenix/db/migrate.py]:
```suggestion
def using_log_level(new_level: int, logger_names: List[str]) -> Iterator[Any]:
```

I don't think `list` works as a generic type on the lowest versions of Python we support.

**Review by axiomofjoy — APPROVED** [review]

File: src/phoenix/db/alembic.ini (hunk 2 above)
Comment (mikeldking) [reply src/phoenix/db/alembic.ini:107]: we can ignore the alembic config if we want but yeah I'd like to be able to set the level overall.

File: src/phoenix/db/migrations/env.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/db/migrations/env.py:18]: That is a weird default.

File: src/phoenix/db/migrate.py (hunk 3 above)
Comment (mikeldking) [reply src/phoenix/db/migrate.py:50]: Agreed, we need a pythonic way to do something like https://www.npmjs.com/package/debug

File: src/phoenix/db/migrate.py (hunk 4 above)
Comment (mikeldking) [reply src/phoenix/db/migrate.py]: Ah yeah good catch

**Pull request merged**